packages/aws-cdk/lib/docker.ts

```diff
@@ -32,83 +22,52 @@ export async function prepareContainerAsset(asset: ContainerImageAssetMetadataEn
                                             toolkitInfo: ToolkitInfo,
                                             reuse: boolean,
                                             ci?: boolean): Promise<CloudFormation.Parameter[]> {
-
   if (reuse) {
     return [
       { ParameterKey: asset.imageNameParameter, UsePreviousValue: true },
     ];
   }
-
   debug(' 👑  Preparing Docker image asset:', asset.path);
-
   const buildHold = new PleaseHold(` ⌛ Building Asset Docker image ${asset.id} from ${asset.path}; this may take a while.`);
   try {
-    const ecr = await toolkitInfo.prepareEcrRepository(asset);
-    const latest = `${ecr.repositoryUri}:latest`;
-
-    let loggedIn = false;
-
-    // In CI we try to pull latest first
-    if (ci) {
-      await dockerLogin(toolkitInfo);
-      loggedIn = true;
-
+    const [ecr, ] = await Promise.all([
+      toolkitInfo.prepareEcrRepository(asset),
+      dockerLogin(toolkitInfo),
+    ]);
+    let command = ['docker', 'build', '--quiet'];
+    const stableTag = crypto.createHash('sha256').update(asset.id).digest('hex');
```

Comment:
Isn't this the same as using `latest`? `prepareEcrRepository` returns a repository with a name derived from `asset.id`:
https://github.com/awslabs/aws-cdk/blob/4e318ed81367b39ee9faa51cba427e5524798db5/packages/aws-cdk/lib/api/toolkit-info.ts#L110-L111

Hashing `asset.id` (a constant for the repository) is for me the same as using a constant (`latest`).

Reply:
No, it is not the same. Using `asset.id` instead of `latest` is the first stepping stone to enable  multiple images per repository.